homeassistant/components/enasolar/sensor.py

```diff
@@ -0,0 +1,288 @@
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_TEMPERATURE,
+    ENERGY_KILO_WATT_HOUR,
+    EVENT_HOMEASSISTANT_START,
+    EVENT_HOMEASSISTANT_STOP,
+    POWER_WATT,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.core import CALLBACK_TYPE, callback
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.event import async_call_later
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    CONF_CAPABILITY,
+    CONF_DC_STRINGS,
+    CONF_MAX_OUTPUT,
+    CONF_SUN_DOWN,
+    CONF_SUN_UP,
+    DEFAULT_SUN_DOWN,
+    DEFAULT_SUN_UP,
+    ENASOLAR_UNIT_MAPPINGS,
+    SCAN_DATA_MIN_INTERVAL,
+    SCAN_MAX_INTERVAL,
+    SCAN_METERS_MIN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the enasolar platform."""
+    _LOGGER.warning(
+        "Configuration of the enasolar platform in configuration.yaml is deprecated "
+        "in Home Assistant 0.119. Please remove entry from your configuration"
+    )
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add enasolar entry."""
+    remove_interval_update_meters = None
+    remove_interval_update_data = None
+
+    # Use all sensors by default
+    hass_meter_sensors = []
+    hass_data_sensors = []
+
+    kwargs = {}
+
+    host = config_entry.data[CONF_HOST]
+
+    try:
+        _LOGGER.info("Instantiate an EnaSolar Inverter at '%s'", host)
+        enasolar = pyenasolar.EnaSolar(host, **kwargs)
+        await enasolar.interogate_inverter()
+    except Exception as e:
+        _LOGGER.error(
+            "Connection to EnaSolar Inverter '%s' failed (%s), check host FQDN/ip address",
+            host,
+            e,
+        )
+        raise PlatformNotReady
+
+    if config_entry.options != {}:
+        enasolar.sun_up = dt_util.parse_time(config_entry.options[CONF_SUN_UP])
+        enasolar.sun_down = dt_util.parse_time(config_entry.options[CONF_SUN_DOWN])
+    else:
+        enasolar.sun_up = dt_util.parse_time(DEFAULT_SUN_UP)
+        enasolar.sun_down = dt_util.parse_time(DEFAULT_SUN_DOWN)
+
+    enasolar.capability = config_entry.data[CONF_CAPABILITY]
+    enasolar.dc_strings = config_entry.data[CONF_DC_STRINGS]
+    enasolar.max_output = config_entry.data[CONF_MAX_OUTPUT]
+
+    _LOGGER.info("Polling between %s and %s", enasolar.sun_up, enasolar.sun_down)
+
+    enasolar.setup_sensors()
+
+    for sensor in enasolar.sensors:
+        _LOGGER.debug("Setup sensor %s", sensor.key)
+        if sensor.enabled:
+            if sensor.is_meter:
+                hass_meter_sensors.append(
+                    EnaSolarSensor(sensor, inverter_name=config_entry.data[CONF_NAME])
+                )
+            else:
+                hass_data_sensors.append(
+                    EnaSolarSensor(sensor, inverter_name=config_entry.data[CONF_NAME])
+                )
+
+    async_add_entities(hass_meter_sensors)
```

Comment:
what is the point of splitting up the list of entities to create in two? The class instances for meter and data sensors seem to be set up the exact same way

Reply:
I wanted to have two different polling frequencies.

Reply:
Oh I see, you are accessing them separately later 👍🏾 

See my comments about the update coordinator. It may not be the right move depending on your use case but I have found it to be pretty flexible and using the update coordinator along with the `CoordinatorEntity` mixin will reduce the amount of boilerplate code.